spharpy/spherical.py

```diff
@@ -1005,3 +1007,413 @@ def sph_identity_matrix(n_max, type='n-nm'):
+        Inverse transform type, either ``'pseudo_inverse'`` or
+        ``'quadrature'``. The default is None.
+    phase_convention : string or None, optional
+        Whether to include the Condon-Shortley phase term.
+        The default is None.
+    """
+
+    def __init__(
+        self,
+        n_max,
+        coordinates,
+        basis_type="real",
+        normalization="n3d",
+        channel_convention="acn",
+        inverse_transform=None,
+        phase_convention=None
+    ):
+        # hidden attributes
+        self._n_max = None
+        self._coordinates = pf.Coordinates()
+        self._basis_type = None
+        self._weights = None
+        self._inverse_transform = None
+        self._channel_convention = None  # ch. ord. conv.
+        self._normalization = None  # gain norm. conv.
+        self._phase_convention = None
+
+        self.n_max = n_max
+        self.coordinates = coordinates
+        self.basis_type = basis_type
+        self.inverse_transform = inverse_transform
+        self.channel_convention = channel_convention
+        self.normalization = normalization
+        self.phase_convention = phase_convention
+
+        self._recompute_basis = True
+        self._recompute_basis_gradient = True
+        self._recompute_inverse = True
+        self._recompute_inverse_gradient = True
+
+        self._basis = None
+        (self._basis_gradient_theta,
+         self._basis_gradient_phi) = (None, None)
+        self._basis_inv = None
+        (self._basis_inv_gradient_theta,
+         self._basis_inv_gradient_phi) = (None, None)
+
+    # Properties
+    # property for phase_convention
+    @property
+    def phase_convention(self):
+        """bool: Get phase convention."""
+        return self._phase_convention
+
+    @phase_convention.setter
+    def phase_convention(self, value):
+        """Set the phase convention."""
+        if not isinstance(value, (str, type(None))):
+            raise TypeError("phase_convention must be a string or None")
+        if value != self._phase_convention:
+            self._recompute_basis = True
+            self._recompute_basis_gradient = True
+            self._recompute_inverse = True
+            self._recompute_inverse_gradient = True
+        self._phase_convention = value
+
+    @property
+    def weights(self):
+        """array-like: Sampling weights for the quadrature transform.
+        Obtained from the coordinates object and are
+        normalized to a sum of 4*pi."""
+        if self._weights is not None:
+            return self._weights
+        elif hasattr(self.coordinates, 'weights') and self.coordinates.weights is not None:
+            return self.coordinates.weights
+        else:
+            # Calculate weights if not already set
+            self._weights = calculate_sampling_weights(self.coordinates)
+            return self._weights
+
+    @weights.setter
+    def weights(self, value):
+        self._weights = value
```

Comment:
If the above is true, we should discuss is we want a separate setter for the weights, or if they have to be set through a Coordinates object

Reply:
If we keep, it should be checked, if value.shape matches the shape of the coordinates